syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.examples.iotservice";
option java_outer_classname = "IoTServiceProto";
option objc_class_prefix = "TEMP";

package temperatura_service;

service TemperaturaService{
    rpc CadastrarTemperatura (Temperatura) returns (StatusReply) {}

    rpc GetTemperaturaPorId (Id) returns (Temperatura) {}

    rpc GetTemperaturaPorData (Data) returns (TemperaturaList) {}

    rpc GetTemperaturaPorLocalizacao (Localizacao) returns (TemperaturaList) {}

    rpc ListarTodasTemperaturas (EmptyMessage) returns (TemperaturaList) {}
}

message Temperatura{
    uint32 id = 1;
    string data = 2;
    string localizacao = 3;
    double temperatura = 4;
}

message Id {
  uint32 id = 1;
}

message StatusReply {
  string status = 1;
}

message Localizacao{
    string localizacao = 1;
}

message Data{
    string data = 1;
}

message EmptyMessage{}

message TemperaturaList{
    repeated Temperatura temperatura_dado = 1;
}